### PR TITLE
WELD-2695 Create smoke test for coexistence of portable extensions with build c…

### PR DESCRIPTION
**Pull request by manovotn**
…ompatible extensions.

### CODE REVIEW

**Comment by manovotn** [comment]
Note that these are just smoke tests and are deliberately running in SE. This is because ATM we rely on CDI TCK to contain tests verifying actual functionality of BCE and those tests are executed via Arq. embedded container where the inclusion of `weld-lite-extension-translator` is done slightly differently.

**Comment by manovotn** [comment]
@Ladicek BTW writing this I realized we don't really have `@SkipIfPortableExtensionPresent` covered in TCKs. Might want to add it as a Full-only test in some simple form as well.

File: environments/se/tests/src/test/java/org/jboss/weld/environment/se/test/extension/build/compatible/OverridenBuildCompatibleExtension.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import jakarta.enterprise.lang.model.declarations.ClassInfo;
+
+/**
+ * This BCE is overriden and should never be invoked
+ */
+// TODO change to OverridingPortableExtension.class once https://github.com/eclipse-ee4j/cdi/issues/585 is resolved
+@SkipIfPortableExtensionPresent("org.jboss.weld.environment.se.test.extension.build.compatible.OverridingPortableExtension")
+public class OverridenBuildCompatibleExtension implements BuildCompatibleExtension {
+
+    public static int TIMES_INVOKED = 0;
```
Comment (mkouba) [inline environments/se/tests/src/test/java/org/jboss/weld/environment/se/test/extension/build/compatible/OverridenBuildCompatibleExtension.java:37]: Just nitpicking over tiny details but are we 100% sure this static variable is only accessed from a single thread?

Comment (manovotn) [reply environments/se/tests/src/test/java/org/jboss/weld/environment/se/test/extension/build/compatible/OverridenBuildCompatibleExtension.java:37]: Yea, we are executing all extensions in a synchronous manner from a single thread, so I'd say it's safe.

Comment (mkouba) [reply environments/se/tests/src/test/java/org/jboss/weld/environment/se/test/extension/build/compatible/OverridenBuildCompatibleExtension.java:37]: If that's the case then we're good to go ;-)

**Review by mkouba — APPROVED** [review]

**Review by Ladicek — APPROVED** [review]
LGTM, I'd just rename `StandardBCE` to `StandardBuildCompatibleExtension`, because all other extensions have their names expanded :-)

**Pull request merged**